### DISCUSSION TITLE
Telemetry for Prometheus query and query_range requests

### DISCUSSION THREAD

**Discussion by dougbrad** [grafana/grafana#48912] [Observability]
There is good telemetry for Prometheus series and series requests through data-proxy-log and context loggers under path /api/v1/series and /api/v1/labels.
However very minimal telemetry for query and query_range calls (In Grafana under path /api/ds/query )

Prometheus requests can't be differentiated under path /api/ds/query, also it appears err info is not populated for /api/ds/query
Only errors are available through logger 'tsdb.prometheus'

Telemetry that is **not** available is:
-   Durations
-   Success's
-   Dashboard uid and name for failures
-   RequestId/TraceId for failures except some cases where in the err 

Also for these calls x-ms-request-id is not set in the response so from HAR file difficult to trace to our logs 

It is important to have telemetry for query and query_range requests as these are the primary requests to query Prometheus.

**Comment by zhichli** [comment]
@taleena This is an ask for improving diagnostic support on Prometheus data sources.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
